fix: test_cli_check_not_in_worktree fails when run from within loom repo

## Bug Description

`TestCLI::test_cli_check_not_in_worktree` in `loom-tools/tests/test_worktree.py` fails when run from within the loom repository itself.

## Reproduction

```bash
cd /path/to/loom  # The loom repo itself
PYTHONPATH=loom-tools/src python3 -m pytest loom-tools/tests/test_worktree.py::TestCLI::test_cli_check_not_in_worktree -v
```

## Failure

```
AssertionError: assert (0 == 1 or 'not' in 'current worktree:\n  path: /users/.../loom\n  branch: docs/guide-update\n')
```

## Root Cause

The test creates a `tmp_path` with a `.git` directory and calls `monkeypatch.chdir(tmp_path)`. However, `main(["--check"])` uses the process's actual `git` context (the loom repo itself, which is a valid git repo and loom worktree), rather than respecting the `chdir`. The `loom-tools` worktree detection apparently reads from the actual git context rather than the CWD.

## Impact

- Test cannot be run from the loom repo (any worktree or main)
- Only passes in isolated environments (CI runs in clean checkout)

## Fix Needed

Either:
1. The worktree detection code should use CWD more consistently, OR
2. The test needs to mock the git discovery mechanism rather than relying on `chdir`

Discovered while evaluating PR #3036.

--- a/loom-tools/tests/test_worktree.py
+++ b/loom-tools/tests/test_worktree.py
@@ -115,16 +115,13 @@ class TestCLI:
         captured = capsys.readouterr()
         assert "does not exist" in captured.err.lower() or "error" in captured.err.lower()
 
-    def test_cli_check_not_in_worktree(self, tmp_path: pathlib.Path, monkeypatch: pytest.MonkeyPatch, capsys: pytest.CaptureFixture[str]) -> None:
-        # Create a simple git repo
-        (tmp_path / ".git").mkdir()
-        monkeypatch.chdir(tmp_path)
-
-        result = main(["--check"])
+    def test_cli_check_not_in_worktree(self, capsys: pytest.CaptureFixture[str]) -> None:
+        """--check reports 'not in a worktree' when _is_in_worktree returns False."""
+        with patch("loom_tools.worktree._is_in_worktree", return_value=False):
+            result = main(["--check"])
         captured = capsys.readouterr()
-
-        # Should indicate not in a worktree
-        assert result == 1 or "not" in captured.out.lower()
+        assert result == 1
+        assert "not" in captured.out.lower()
 
 
 class TestDocumentedBehaviorDifferences:

